src/second_ode_solve.jl

```diff
@@ -0,0 +1,58 @@
+struct NNODE2{C,O} <: NeuralNetDiffEqAlgorithm
+    chain::C
+    opt::O
+end
+NNODE2(chain;opt=Flux.ADAM(0.1)) = NNODE2(chain,opt)
+
+function DiffEqBase.solve(
+    prob::DiffEqBase.AbstractODEProblem,
+    alg::NNODE2,
+    args...;
+    dt,
+    timeseries_errors = true,
+    save_everystep=true,
+    adaptive=false,
+    abstol = 1f-6,
+    verbose = false,
+    maxiters = 100)
+
+    u0, du0 = prob.u0
+    tspan = prob.tspan
+    f = prob.f.f1
+    p = prob.p
+    t0 = tspan[1]
+
+    #hidden layer
+    chain  = alg.chain
+    opt    = alg.opt
+    ps     = Flux.params(chain)
+    data   = Iterators.repeated((), maxiters)
+
+    #train points generation
+    cnt = 0.0f0
+    ts  = tspan[1]:dt:tspan[2]
+    
+    #Initial Value Problem
+    phi(t) = (u0 .+ du0 .* (t .- tspan[1]) .+ (t .- tspan[1]) .^ 2 .* chain([t]))[1]
+
+
+    dfdx = t -> (phi(t+sqrt(eps(typeof(dt)))) - phi(t)) / sqrt(eps(typeof(dt)))
+    epsilon(t) = cbrt(eps(typeof(dt)))
+    #second order central
+    d2fdx2(t) = (phi(t+epsilon(t)) - 2phi(t) + phi(t-epsilon(t)))/epsilon(t)^2
+    loss = () -> sum(abs2,sum(abs2,d2fdx2(t) .- f(dfdx(t), phi(t),p,t)[1]) for t in ts)
```

Comment:
add support for systems of equations here. You can see how this is done in ode_solver `if u0 isa Number .. else` and etc

Reply:
probably, we can avoid using `if...else`. Anyway, it is needed to be fixed. We get `boundary Error` for the example from `NNODE2_tests2.jl`